### PR TITLE
[WIP] ES6: Convert discussion/comments

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

ES6: Converts discussion/comments

### CODE REVIEW

File: static/src/javascripts/lib/$.js [hunk 1]
```diff
@@ -24,9 +24,11 @@ $.ancestor = (el: ?Node, className: string): ?Node => {
     ) {
         return null;
     }
+
     if (!el.parentNode || bonzo(el.parentNode).hasClass(className)) {
         return el.parentNode;
     }
+
```
Comment (SiAdcock) [inline static/src/javascripts/lib/$.js]: Can we kill the whitespace changes to this file

**Comment by SiAdcock** [comment]
Don't forget to delete the legacy version of this file, and remove its entry from `es5to6.json`!

File: static/src/javascripts/projects/common/modules/discussion/comments.js [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+import { constructQuery } from 'lib/url';
+import Component from 'common/modules/component';
+import DiscussionApi from 'common/modules/discussion/api';
+import CommentBox from 'common/modules/discussion/comment-box';
+import WholeDiscussion from 'common/modules/discussion/whole-discussion';
+import relativedates from 'common/modules/ui/relativedates';
+import userPrefs from 'common/modules/user-prefs';
+import { inlineSvg } from 'common/views/svgs';
+
+const PREF_RELATIVE_TIMESTAMPS: string = 'discussion.enableRelativeTimestamps';
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/comments.js]: Is the type annotation here useful?

**Comment by SiAdcock** [comment]
It looks like the [`ready` method](https://github.com/guardian/frontend/blob/master/static/src/javascripts-legacy/projects/common/modules/discussion/comments.js#L85-L109) has gone missing during this conversion. Is that expected?

File: static/src/javascripts/projects/common/modules/discussion/comments.js [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+            })
+            .catch(ex => {
+                reportError(ex, {
+                    feature: 'comments-more-replies',
+                });
+            });
+    }
+
+    addMoreRepliesButtons(comms: qwery): void {
+        const comments = comms || this.topLevelComments;
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/comments.js]: Could `comms` ever be falsey?

File: static/src/javascripts/projects/common/modules/discussion/comments.js [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+                        ? JSON.parse(resp.response).message
+                        : resp.statusText;
+
+                $(target).text(responseText);
+            });
+        }
+    }
+
+    comments: ?qwery;
+    topLevelComments: ?qwery;
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/comments.js]: Is it just me, or is it odd that these class properties appear halfway down the class definition?

File: static/src/javascripts/projects/common/modules/discussion/comments.js [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+        if (
+            showRepliesElem.length > 0 &&
+            !bonzo(showRepliesElem).hasClass('u-h')
+        ) {
+            showRepliesElem[0].click();
+        }
+
+        commentBox.render(parentCommentEl);
+
+        // TODO (jamesgorrie): Remove Hack hack hack
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/comments.js]: This `TODO` is totally legit

File: static/src/javascripts/lib/$.js (hunk 1 above)
Comment (gustavpursche) [inline static/src/javascripts/lib/$.js]: Sure.

File: static/src/javascripts/projects/common/modules/discussion/comments.js (hunk 2 above)
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/discussion/comments.js]: No, but also not harmful :)

File: static/src/javascripts/projects/common/modules/discussion/comments.js (hunk 4 above)
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/discussion/comments.js]: Eslint forced me to do so :/

**Comment by gustavpursche** [comment]
> It looks like the ready method has gone missing during this conversion. Is that expected?

Nice catch. Thanks! 🌷 

> Don't forget to delete the legacy version of this file, and remove its entry from es5to6.json!

I just realized that I removed the file - but in a different branch. 👀

**Pull request closed without merging**